lib/diagnostic.js

```diff
@@ -29,14 +43,30 @@ const newArray = function newArray(length, defaultValue) {
 // to the result of invoking defaultValue with the
 // element's index as an argument
 
+const newArray = function newArray(length, defaultValue) {
+  var array = [];
+  for(i = 0; i<length; i++){
+    if(typeof defaultValue === function){
+    array[i] = defaultValue(i);
+    } else {
+    array[i] = defaultValue;
+    }
+  }
+  return array;
+};
```

Comment:
This needs to use the `const newArray = ` variable. Since you reassigned it, it breaks the tests. Be sure to watch out for reassigned, or trying to, in the future.